src/mono/mono/mini/interp/interp.c

```diff
@@ -6065,19 +6065,31 @@ MINT_IN_CASE(MINT_BRTRUE_I8_SP) ZEROP_SP(gint64, !=); MINT_IN_BREAK;
 			int index = LOCAL_VAR (ip [3], int);
 			NULL_CHECK (span);
 
-			gsize offset_length = (gsize)(gint16)ip [5];
-
-			const gint32 length = *(gint32 *) (span + offset_length);
+			const gint32 length = *(gint32 *) (span + TARGET_SIZEOF_VOID_P);
```

Comment:
There is a MonoSpanOfObjects struct which can be used instead of these pointer additions.

Reply:
Added a new MonoSpanOfVoid so there is no confusion about elements being objects here.